### PR TITLE
Create music album and genre class

### PR DESCRIPTION
**Pull request by koomekevo**
This Pull Request has been created to achieve the following: 

- Create MusicAlbum class in a separate .rb file. 
- Add and Implement methods in MusicAlbum class.
- [ ] can_be_archived?() in the MusicAlbum class
- should override the method from the parent class
- should return true if parent's method returns true AND if on_spotify equals true
- otherwise, it should return false

- Create Genre class with an association to the Item class (in a separate .rb file).
- All Genre class properties visible in the diagram should be defined and set up in the constructor method.
- [ ] add_item method in the Genre class
- should take an instance of the Item class as an input
- should add the input item to the collection of items
- should add self as a property of the item object (by using the correct setter from the item object)

### CODE REVIEW

File: music_album.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative 'item'
+
+class MusicAlbum < Item
+  attr_accessor :on_spotify
+
+  def initialize(publish_date, on_spotify)
+    super(id, name, genre, label, publish_date)
```
Comment (ReAliens) [inline music_album.rb:7]: here, please call only included arguments in initialize method in super constructor

**Review by ReAliens — CHANGES_REQUESTED** [review]
Hello @koomekevo 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### **Highlights**
- PR is good :ok_hand: 
- Linters are passing :+1: 
**### Status: Required Changes**
Check the comments under the review.

### **Optional suggestions**
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:

N/A
Cheers and Happy coding! :clap: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: genre.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Genre
+  attr_accessor :name, :id, :items
```
Comment (ReAliens) [inline genre.rb:2]: - please we don't need to access name and id here as a genre has its own id and name
- you should import item class to access it

File: genre.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(name, id)
+    @id = id
+    @name = name
+    @items = items[]
+  end
+
+  def add_item(item)
+    @items << item
+    item.genre = self
```
Comment (ReAliens) [inline genre.rb:12]: I don't think that genre is a property of an item but it's a method inside it so please write a correct syntax and pass self notation to it as an argument :ok_hand:

File: music_album.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require_relative 'item'
+
+class MusicAlbum < Item
+  attr_accessor :on_spotify
```
Comment (ReAliens) [inline music_album.rb:4]: we don't need access here as you are asking it in initialize method

**Review by ReAliens — CHANGES_REQUESTED** [review]
Hello @koomekevo 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### **Highlights**
- PR is good :ok_hand: 
- Linters are passing :+1: 

### **Status: Required Changes**
Check the comments under the review.

### **Optional suggestions**
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:

N/A
Cheers and Happy coding! :clap: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

**Review by ReAliens — APPROVED** [review]
Hi @koomekevo 👋🏽,
done

Everything is working fine now :wink: 
Your project is complete!
There is nothing else to say other than... it's time to merge it :+1: :sunglasses: 

Congratulations! :1st_place_medal: 
Happy coding :clap: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by AnselemOdims** [comment]
Good job guys.

**Pull request merged**